packages/vats/src/vat-agoricNames.js

```diff
@@ -0,0 +1,92 @@
+// @ts-check
+// @jessy-check
+import { E, Far } from '@endo/far';
+import { BrandI } from '@agoric/ertp';
+import { provide } from '@agoric/vat-data';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { prepareNameHubKit } from './nameHub.js';
+
+/** @param {import('@agoric/zone').Zone} zone */
+const makeBrandStore = zone => {
+  const brandStore = zone.mapStore('Brand');
+
+  // XXX generalize past Nat; move into ERTP
+  /** @type {(name: string, d: DisplayInfo) => Brand} */
+  const makeNatBrand = zone.exoClass(
+    'Brand',
+    BrandI,
+    (name, displayInfo) => ({ name, displayInfo }),
+    {
+      async isMyIssuer(_allegedIssuer) {
+        return false;
+      },
+      getAllegedName() {
+        const { name } = this.state;
+        return name;
+      },
+      // Give information to UI on how to display the amount.
+      getDisplayInfo() {
+        const { displayInfo } = this.state;
+        return displayInfo;
+      },
+      getAmountShape() {
+        return undefined;
+      },
+    },
+  );
+
+  return {
+    provide: (keyword, displayInfo) =>
```

Comment:
The plain `provide` is for when what is provided is specified by the `make` argument. In this case it's always making a brand (and that's not implied by the signature).
```suggestion
    provideBrand: (keyword, displayInfo) =>
```